packages/flutter/lib/src/rendering/sliver.dart

```diff
@@ -450,20 +451,21 @@ class SliverConstraints extends Constraints {
   bool operator ==(dynamic other) {
     if (identical(this, other))
       return true;
-    if (other is! SliverConstraints)
-      return false;
-    final SliverConstraints typedOther = other;
-    assert(typedOther.debugAssertIsValid());
-    return typedOther.axisDirection == axisDirection
-        && typedOther.growthDirection == growthDirection
-        && typedOther.scrollOffset == scrollOffset
-        && typedOther.overlap == overlap
-        && typedOther.remainingPaintExtent == remainingPaintExtent
-        && typedOther.crossAxisExtent == crossAxisExtent
-        && typedOther.crossAxisDirection == crossAxisDirection
-        && typedOther.viewportMainAxisExtent == viewportMainAxisExtent
-        && typedOther.remainingCacheExtent == remainingCacheExtent
-        && typedOther.cacheOrigin == cacheOrigin;
+    assert(() {
```

Comment:
This change doesn't seem to preserve the old logic. Previously, if `other` was not a `SliverConstraints` it would return false. Now, the assert would throw, no?

Reply:
Good catch!